src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> nextNode;
+        private Node<T> previousNode;
+
+        public Node(Node<T> previousNode, T value, Node<T> nextNode) {
+            this.nextNode = nextNode;
+            this.value = value;
+            this.previousNode = previousNode;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            Node<T> newNode = new Node<>(null, value, null);
+            head = newNode;
+            tail = newNode;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.nextNode = newNode;
+            tail = newNode;
+
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        indexCheckForAdd(index);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> insertingNode = new Node<>(null, value, head);
+            head = insertingNode;
+            size++;
+        } else {
+            Node<T> insertingNode;
+            Node newNode = getNodeNyIndex(index - 1);
+            insertingNode = new Node<>(newNode, value, newNode.nextNode);
+            Node afteradditingNode = newNode.nextNode;
+            newNode.nextNode = insertingNode;
+            insertingNode.previousNode = newNode;
```

Comment:
we can remove this line (we do it via constructor)

Reply:
done